src/map.h

```diff
@@ -1052,6 +1067,20 @@ void add_corpse(int x, int y);
          * tr_null trap.
          */
         std::vector< std::vector<tripoint> > traplocs;
+
+    // cached values for map visibility calculations
+    int g_light_level;
```

Comment:
Why are these here?  The original method local caching was a no brainer, but I can't say the same for embedding them in the class and persisting them like this does.
If you're concerned about calling the accessors for these too many times, why did you break up map::draw()?